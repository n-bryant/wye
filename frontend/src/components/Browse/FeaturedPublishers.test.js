import { act } from "react-dom/test-utils";
import { MockedProvider } from "@apollo/react-testing";
import wait from "waait";

import FeaturedPublishersQuery, {
  FeaturedPublishers,
  StyledFeaturedPublishers,
  GET_FEATURED_PUBLISHERS_TOP_TITLE_QUERY
} from "./FeaturedPublishers";

const mockedResult = {
  getTopTitleForMostPopularPublishers: [
    {
      publisher: "1",
      topTitle: {
        appid: "1",
        name: "name",
        developers: ["a"],
        publishers: ["a"],
        ownersFormatted: "10,000 .. 20,000",
        userRating: 95,
        genres: ["a"],
        freeToPlay: false,
        onSale: true,
        discount: 25,
        initialPrice: 1000,
        finalPrice: 750,
        headerImage: "header",
        capsuleMd: "md"
      }
    },
    {
      publisher: "2",
      topTitle: {
        appid: "2",
        name: "name",
        developers: ["a"],
        publishers: ["a"],
        ownersFormatted: "10,000 .. 20,000",
        userRating: 95,
        genres: ["a"],
        freeToPlay: false,
        onSale: true,
        discount: 25,
        initialPrice: 1000,
        finalPrice: 750,
        headerImage: "header",
        capsuleMd: "md"
      }
    },
    {
      publisher: "3",
      topTitle: {
        appid: "3",
        name: "name",
        developers: ["a"],
        publishers: ["a"],
        ownersFormatted: "10,000 .. 20,000",
        userRating: 95,
        genres: ["a"],
        freeToPlay: false,
        onSale: true,
        discount: 25,
        initialPrice: 1000,
        finalPrice: 750,
        headerImage: "header",
        capsuleMd: "md"
      }
    },
    {
      publisher: "4",
      topTitle: {
        appid: "4",
        name: "name",
        developers: ["a"],
        publishers: ["a"],
        ownersFormatted: "10,000 .. 20,000",
        userRating: 95,
        genres: ["a"],
        freeToPlay: false,
        onSale: true,
        discount: 25,
        initialPrice: 1000,
        finalPrice: 750,
        headerImage: "header",
        capsuleMd: "md"
      }
    }
  ]
};

describe("FeaturedPublishersQuery", () => {
  it("should render successfully", () => {
    const mocks = [
      {
        request: {
          query: GET_FEATURED_PUBLISHERS_TOP_TITLE_QUERY
        },
        result: {
          data: mockedResult
        }
      }
    ];
    const wrapper = mount(
      <MockedProvider mocks={mocks} addTypename={false}>
        <FeaturedPublishersQuery />
      </MockedProvider>
    );
    expect(toJson(wrapper.find(FeaturedPublishersQuery))).toMatchSnapshot();
  });

  it("should return a StyledFeaturedPublishers", async () => {
    const mocks = [
      {
        request: {
          query: GET_FEATURED_PUBLISHERS_TOP_TITLE_QUERY
        },
        result: {
          data: mockedResult
        }
      }
    ];
    const wrapper = mount(
      <MockedProvider mocks={mocks} addTypename={false}>
        <FeaturedPublishersQuery />
      </MockedProvider>
    );

    await act(async () => {
      await wait(0); // wait for response
      wrapper.update();
    });
    expect(wrapper.find(StyledFeaturedPublishers).length).toBe(1);
  });
});

describe("FeaturedPublishers", () => {
  const props = {
    classes: {
      root: "root",
      itemsContainer: "itemsContainer",
      item: "item"
    },
    items: mockedResult.getTopTitleForMostPopularPublishers,
    width: "xs"
  };

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<FeaturedPublishers {...props} />);
  });

  it("should render successfully", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should adjust the layout of the cards if the width is larger than xs or sm", () => {
    wrapper.setProps({
      width: "md"
    });
    wrapper.update();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
