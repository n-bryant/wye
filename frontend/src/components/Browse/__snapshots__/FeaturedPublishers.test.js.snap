// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedPublishers should adjust the layout of the cards if the width is larger than xs or sm 1`] = `
<WithStyles(FeaturedWidget)
  className="wye-FeaturedPublishers root"
  classes={Object {}}
  subTitle="The most played titles by the most prolific publishers"
  title="Publishers"
>
  <WithStyles(ForwardRef(Grid))
    className="wye-FeaturedPublishers__itemsContainer itemsContainer"
    container={true}
    justify="center"
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      className="wye-FeaturedPublishers__item item"
      item={true}
      key="1"
      xs={3}
    >
      <HighlightTrailerQuery
        cardActionLabel="1"
        cardActionLinkPath="/browse/publisher/1"
        data={
          Object {
            "appid": "1",
            "capsuleMd": "md",
            "developers": Array [
              "a",
            ],
            "discount": 25,
            "finalPrice": 750,
            "freeToPlay": false,
            "genres": Array [
              "a",
            ],
            "headerImage": "header",
            "initialPrice": 1000,
            "name": "name",
            "onSale": true,
            "ownersFormatted": "10,000 .. 20,000",
            "publishers": Array [
              "a",
            ],
            "userRating": 95,
          }
        }
        variant="md"
        withPrice={false}
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      className="wye-FeaturedPublishers__item item"
      item={true}
      key="2"
      xs={3}
    >
      <HighlightTrailerQuery
        cardActionLabel="2"
        cardActionLinkPath="/browse/publisher/2"
        data={
          Object {
            "appid": "2",
            "capsuleMd": "md",
            "developers": Array [
              "a",
            ],
            "discount": 25,
            "finalPrice": 750,
            "freeToPlay": false,
            "genres": Array [
              "a",
            ],
            "headerImage": "header",
            "initialPrice": 1000,
            "name": "name",
            "onSale": true,
            "ownersFormatted": "10,000 .. 20,000",
            "publishers": Array [
              "a",
            ],
            "userRating": 95,
          }
        }
        variant="md"
        withPrice={false}
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      className="wye-FeaturedPublishers__item item"
      item={true}
      key="3"
      xs={3}
    >
      <HighlightTrailerQuery
        cardActionLabel="3"
        cardActionLinkPath="/browse/publisher/3"
        data={
          Object {
            "appid": "3",
            "capsuleMd": "md",
            "developers": Array [
              "a",
            ],
            "discount": 25,
            "finalPrice": 750,
            "freeToPlay": false,
            "genres": Array [
              "a",
            ],
            "headerImage": "header",
            "initialPrice": 1000,
            "name": "name",
            "onSale": true,
            "ownersFormatted": "10,000 .. 20,000",
            "publishers": Array [
              "a",
            ],
            "userRating": 95,
          }
        }
        variant="md"
        withPrice={false}
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      className="wye-FeaturedPublishers__item item"
      item={true}
      key="4"
      xs={3}
    >
      <HighlightTrailerQuery
        cardActionLabel="4"
        cardActionLinkPath="/browse/publisher/4"
        data={
          Object {
            "appid": "4",
            "capsuleMd": "md",
            "developers": Array [
              "a",
            ],
            "discount": 25,
            "finalPrice": 750,
            "freeToPlay": false,
            "genres": Array [
              "a",
            ],
            "headerImage": "header",
            "initialPrice": 1000,
            "name": "name",
            "onSale": true,
            "ownersFormatted": "10,000 .. 20,000",
            "publishers": Array [
              "a",
            ],
            "userRating": 95,
          }
        }
        variant="md"
        withPrice={false}
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(FeaturedWidget)>
`;

exports[`FeaturedPublishers should render successfully 1`] = `
<WithStyles(FeaturedWidget)
  className="wye-FeaturedPublishers root"
  classes={Object {}}
  subTitle="The most played titles by the most prolific publishers"
  title="Publishers"
>
  <WithStyles(ForwardRef(Grid))
    className="wye-FeaturedPublishers__itemsContainer itemsContainer"
    container={true}
    justify="center"
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      className="wye-FeaturedPublishers__item item"
      item={true}
      key="1"
      xs={6}
    >
      <HighlightTrailerQuery
        cardActionLabel="1"
        cardActionLinkPath="/browse/publisher/1"
        data={
          Object {
            "appid": "1",
            "capsuleMd": "md",
            "developers": Array [
              "a",
            ],
            "discount": 25,
            "finalPrice": 750,
            "freeToPlay": false,
            "genres": Array [
              "a",
            ],
            "headerImage": "header",
            "initialPrice": 1000,
            "name": "name",
            "onSale": true,
            "ownersFormatted": "10,000 .. 20,000",
            "publishers": Array [
              "a",
            ],
            "userRating": 95,
          }
        }
        variant="md"
        withPrice={false}
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      className="wye-FeaturedPublishers__item item"
      item={true}
      key="2"
      xs={6}
    >
      <HighlightTrailerQuery
        cardActionLabel="2"
        cardActionLinkPath="/browse/publisher/2"
        data={
          Object {
            "appid": "2",
            "capsuleMd": "md",
            "developers": Array [
              "a",
            ],
            "discount": 25,
            "finalPrice": 750,
            "freeToPlay": false,
            "genres": Array [
              "a",
            ],
            "headerImage": "header",
            "initialPrice": 1000,
            "name": "name",
            "onSale": true,
            "ownersFormatted": "10,000 .. 20,000",
            "publishers": Array [
              "a",
            ],
            "userRating": 95,
          }
        }
        variant="md"
        withPrice={false}
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    className="wye-FeaturedPublishers__itemsContainer itemsContainer"
    container={true}
    justify="center"
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      className="wye-FeaturedPublishers__item item"
      item={true}
      key="3"
      xs={6}
    >
      <HighlightTrailerQuery
        cardActionHref="/browse/publisher/[id]"
        cardActionLabel="3"
        cardActionLinkPath="/browse/publisher/3"
        data={
          Object {
            "appid": "3",
            "capsuleMd": "md",
            "developers": Array [
              "a",
            ],
            "discount": 25,
            "finalPrice": 750,
            "freeToPlay": false,
            "genres": Array [
              "a",
            ],
            "headerImage": "header",
            "initialPrice": 1000,
            "name": "name",
            "onSale": true,
            "ownersFormatted": "10,000 .. 20,000",
            "publishers": Array [
              "a",
            ],
            "userRating": 95,
          }
        }
        variant="md"
        withPrice={false}
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      className="wye-FeaturedPublishers__item item"
      item={true}
      key="4"
      xs={6}
    >
      <HighlightTrailerQuery
        cardActionHref="/browse/publisher/[id]"
        cardActionLabel="4"
        cardActionLinkPath="/browse/publisher/4"
        data={
          Object {
            "appid": "4",
            "capsuleMd": "md",
            "developers": Array [
              "a",
            ],
            "discount": 25,
            "finalPrice": 750,
            "freeToPlay": false,
            "genres": Array [
              "a",
            ],
            "headerImage": "header",
            "initialPrice": 1000,
            "name": "name",
            "onSale": true,
            "ownersFormatted": "10,000 .. 20,000",
            "publishers": Array [
              "a",
            ],
            "userRating": 95,
          }
        }
        variant="md"
        withPrice={false}
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(FeaturedWidget)>
`;

exports[`FeaturedPublishersQuery should render successfully 1`] = `
<FeaturedPublishersQuery>
  <Query
    query={
      Object {
        "definitions": Array [
          Object {
            "directives": Array [],
            "kind": "OperationDefinition",
            "name": Object {
              "kind": "Name",
              "value": "GET_FEATURED_PUBLISHERS_TOP_TITLE_QUERY",
            },
            "operation": "query",
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "getTopTitleForMostPopularPublishers",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "publisher",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "topTitle",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "appid",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "name",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "developers",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "publishers",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "ownersFormatted",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "userRating",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "genres",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "freeToPlay",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "onSale",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "discount",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "initialPrice",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "finalPrice",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "headerImage",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "capsuleMd",
                              },
                              "selectionSet": undefined,
                            },
                          ],
                        },
                      },
                    ],
                  },
                },
              ],
            },
            "variableDefinitions": Array [],
          },
        ],
        "kind": "Document",
        "loc": Object {
          "end": 396,
          "start": 0,
        },
      }
    }
  >
    <WithStyles(LoadingState)
      classes={Object {}}
    >
      <LoadingState
        classes={
          Object {
            "progressIndicator": "LoadingState-progressIndicator-2",
            "root": "LoadingState-root-1",
          }
        }
      >
        <div
          className="wye-LoadingState LoadingState-root-1"
        >
          <WithStyles(ForwardRef(CircularProgress))
            className="wye-LoadingState__progressIndicator LoadingState-progressIndicator-2"
          >
            <ForwardRef(CircularProgress)
              className="wye-LoadingState__progressIndicator LoadingState-progressIndicator-2"
              classes={
                Object {
                  "circle": "MuiCircularProgress-circle",
                  "circleDisableShrink": "MuiCircularProgress-circleDisableShrink",
                  "circleIndeterminate": "MuiCircularProgress-circleIndeterminate",
                  "circleStatic": "MuiCircularProgress-circleStatic",
                  "colorPrimary": "MuiCircularProgress-colorPrimary",
                  "colorSecondary": "MuiCircularProgress-colorSecondary",
                  "indeterminate": "MuiCircularProgress-indeterminate",
                  "root": "MuiCircularProgress-root",
                  "static": "MuiCircularProgress-static",
                  "svg": "MuiCircularProgress-svg",
                }
              }
            >
              <div
                className="MuiCircularProgress-root wye-LoadingState__progressIndicator LoadingState-progressIndicator-2 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
                role="progressbar"
                style={
                  Object {
                    "height": 40,
                    "width": 40,
                  }
                }
              >
                <svg
                  className="MuiCircularProgress-svg"
                  viewBox="22 22 44 44"
                >
                  <circle
                    className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                    cx={44}
                    cy={44}
                    fill="none"
                    r={20.2}
                    strokeWidth={3.6}
                    style={Object {}}
                  />
                </svg>
              </div>
            </ForwardRef(CircularProgress)>
          </WithStyles(ForwardRef(CircularProgress))>
        </div>
      </LoadingState>
    </WithStyles(LoadingState)>
  </Query>
</FeaturedPublishersQuery>
`;
